File: cpp/mahasiswa.cpp
//import library
#include <bits/stdc++.h>
#include "sivitas.cpp"

//using standard namespace
using namespace std;

class Mahasiswa : public SivitasAkademik
{
    private:
        string nim;
        string nama;
        char jenis_kelamin;
        string fakultas;
        string prodi;

    public:
        Mahasiswa()
        {
            this->nim = "";
            this->nama = "";
            this->jenis_kelamin = '-';
            this->fakultas = "";
            this->prodi = "";
        }

        string get_nim()
        {
            return this->nim;
        }

        void set_nim(string nim)
        {
            this->nim = nim;
        }

        string get_nama()
        {
            return this->nama;
        }

        void set_nama(string nama)
        {
            this->nama = nama;
        }

        char get_jenis_kelamin()
        {
            return this->jenis_kelamin;
        }

        void set_jenis_kelamin(char jenis_kelamin)
        {
            this->jenis_kelamin = jenis_kelamin;
        }

        string get_fakultas()
        {
            return this->fakultas;
        }

        void set_fakultas(string fakultas)
        {
            this->fakultas = fakultas;
        }

        string get_prodi()
        {
            return this->prodi;
        }

        void set_prodi(string prodi)
        {
            this->prodi = prodi;
        }

        ~Mahasiswa()
        {

        }

};

File: mahasiswa.java
//Kelas SivitasAkademik menjadi induk Mahasiswa
class Mahasiswa extends SivitasAkademik
{
    //private
    private String nim;
    private String nama;
    private String jenis_kelamin;
    private String fakultas;
    private String prodi;

    //public
    public Mahasiswa()
    {
        this.nim = "";
        this.nama = "";
        this.jenis_kelamin = "";
        this.fakultas = "";
        this.prodi = "";
    }

    //getter setter
    public String getnim()
    {
        return this.nim;
    }

    public void setnim(String nim)
    {
        this.nim = nim;
    }
        
    public String getnama()
    {
        return this.nama;
    }
    
    public void setnama(String nama)
    {
        this.nama = nama;
    }
    public String getjenis_kelamin()
    {
        return this.jenis_kelamin;
    }
    
    public void setjenis_kelamin(String jenis_kelamin)
    {
        this.jenis_kelamin = jenis_kelamin;
    }
    public String getfakultas()
    {
        return this.fakultas;
    }
    
    public void setfakultas(String fakultas)
    {
        this.fakultas = fakultas;
    }
    public String getprodi()
    {
        return this.prodi;
    }
    
    public void setprodi(String prodi)
    {
        this.prodi = prodi;
    }
}